packages/echo-core/src/__tests__/actions/legendOptionsStateActions.test.ts

```diff
@@ -1,49 +1,85 @@
+import { eventHub } from '@equinor/echo-base';
 import { dispatch, readState } from '../../actions/coreActions/globalActions';
 import { getLegendOption, setLegendOption } from '../../actions/legendOptions';
 import { defaultGlobalState } from '../../state/defaultStates';
 import { getCoreContext } from '../../state/globalState';
-import { GlobalState } from '../../types';
-
-beforeEach(() => {
-    initialize();
-});
+import { EchoEvents, GlobalState } from '../../types';
 
 function initialize(): void {
     dispatch(getCoreContext(), () => defaultGlobalState);
 }
 
 describe('legendOptionsStateActions', () => {
-    const expectedLegendOptions = {
-        isActive: false,
-        selectedLegendType: 'test'
-    };
-
+    beforeEach(() => {
+        initialize();
+    });
     describe('setLegendOption', () => {
-        it('should set legend option state', () => {
-            setLegendOption({ isActive: false, selectedLegendType: 'test' });
-            expect(expectedLegendOptions).toEqual(
-                readState(getCoreContext(), (state: GlobalState) => state.legendOptions)
-            );
+        it('should set legend option state and dispatch a legendTypeChanged event', () => {
+            // given
+            let actualEventHubPayload;
+            const legendOptionToSet = { isActive: false, selectedLegendType: 'testLegendType' };
+            const unsubscribe = eventHub.subscribe(EchoEvents.LegendTypeChanged, (payload) => {
+                actualEventHubPayload = payload;
+            });
+
+            // when
+            setLegendOption(legendOptionToSet);
+
+            // then
+            expect(legendOptionToSet).toEqual(readState(getCoreContext(), (state: GlobalState) => state.legendOptions));
+            expect(actualEventHubPayload).toEqual({
+                newLegendType: 'testLegendType'
+            });
+            unsubscribe();
         });
 
-        it('should show default global state', () => {
-            expectedLegendOptions.isActive = true;
-            expectedLegendOptions.selectedLegendType = 'Stid';
+        it('should set legend option state when only setting `isActive` and not emit `legendTypeChanged` event ', () => {
```

Comment:
I was a bit confused with this description
if I understand this correctly, maybe this is more clear: should not emit `legendTypeChanged` event when `isActive` is false

Reply:
I must agree, reading it again it makes me also confused a bit :,D I try to rephrase it:
`should not emit legendTypeChanged if legendType hasn't been changed`